src/main/java/com/tutorhub/model/course/Progress.java

```diff
@@ -2,30 +2,54 @@
 
 import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
-import jakarta.persistence.FetchType;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
 import jakarta.persistence.JoinColumn;
-import jakarta.persistence.OneToMany;
+import jakarta.persistence.ManyToOne;
 import jakarta.persistence.Table;
 import lombok.Getter;
 import lombok.Setter;
 
-import java.util.List;
-
 @Entity
 @Table(name = "progress")
 @Getter
 @Setter
+@IdClass(ProgressId.class)
 public class Progress {
 
   @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  private Long id;
+  @Column(name = "module_id", nullable = false)
+  private Long moduleID;
+
+  @Id
+  @Column(name = "submodule_id", nullable = false)
+  private Long submoduleID;
+
+  @Id
+  @Column(name = "course_info_id", nullable = false)
+  private Long courseInfoID;
+
+  @ManyToOne
+  @JoinColumn(
+      name = "module_id",
+      referencedColumnName = "id",
+      insertable = false,
+      updatable = false)
+  private Module module;
+
+  @ManyToOne
+  @JoinColumn(
+      name = "submodule_id",
+      referencedColumnName = "id",
+      insertable = false,
+      updatable = false)
+  private Submodule submodule;
 
-  @Column(name = "submodule_id")
-  @OneToMany(fetch = FetchType.EAGER)
-  @JoinColumn(name = "submodule_id")
-  private List<Submodule> passedModules;
+  @ManyToOne
+  @JoinColumn(
+      name = "course_info_id",
+      referencedColumnName = "id",
+      insertable = false,
+      updatable = false)
+  private CourseInfo courseInfo;
```

Comment:
Why do you have `moduleId` and `module`, `submoduleId` and `submodule` and `courseInfoId` and `courseInfo` in the same class?

Reply:
I thought the composite key wouldn't work without those fields, but now that I've removed the extra fields, it works

Reply:
Fixed